api/src/main/java/org/pmiops/workbench/actionaudit/auditors/ClusterAuditorImpl.java

```diff
@@ -0,0 +1,47 @@
+package org.pmiops.workbench.actionaudit.auditors;
+
+import org.pmiops.workbench.actionaudit.ActionAuditEvent;
+import org.pmiops.workbench.actionaudit.ActionAuditService;
+import org.pmiops.workbench.actionaudit.ActionType;
+import org.pmiops.workbench.actionaudit.AgentType;
+import org.pmiops.workbench.actionaudit.TargetType;
+import org.pmiops.workbench.actionaudit.targetproperties.AccountTargetProperty;
+import org.pmiops.workbench.actionaudit.targetproperties.values.AccountDisabledStatus;
+import org.pmiops.workbench.db.model.DbUser;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Service;
+
+import javax.inject.Provider;
+import java.time.Clock;
+
+@Service
+public class ClusterAuditorImpl implements ClusterAuditor {
+  private ActionAuditService actionAuditService;
+  private Clock clock;
+  private Provider<DbUser> adminDbUserProvider;
+
+  @Autowired
+  public ClusterAuditorImpl(
+      ActionAuditService actionAuditService,
+      Clock clock,
+      Provider<DbUser> adminDbUserProvider) {
```

Comment:
Nitpick: I'm not sure the "admin" prefix is too helpful here. We have a pretty strong pattern of calling this member variable "userProvider". I see that locally within this package AuthDomainAuditorImpl also calls it "adminDbUserProvider" (which is probably where this naming came from), but I'd argue both usages are inconsistent with a pretty global pattern across our codebase. It crease more confusion than it resolves IMO, since the "user" is still the "user from the current request context" in all cases.